packages/apm/src/hubextensions.ts

```diff
@@ -26,43 +27,50 @@ function traceHeaders(): { [key: string]: string } {
  *
  * @param spanContext Properties with which the span should be created
  */
-function startSpan(spanContext?: SpanContext): Span {
+function startSpan(spanOrTransactionContext: SpanContext | TransactionContext): Transaction | Span {
   // @ts-ignore
   const hub = this as Hub;
   const scope = hub.getScope();
   const client = hub.getClient();
-  let span;
 
-  // This flag determines if we already added the span as a child to the span that currently lives on the scope
-  // If we do not have this, we will add it later on twice to the span recorder and therefore have too many spans
-  let addedAsChild = false;
+  let newSpanContext = spanOrTransactionContext;
 
+  // We create an empty Span in case there is no scope on the hub
+  let parentSpan = new Span();
   if (scope) {
-    const parentSpan = scope.getSpan() as Span;
+    // If there is a Span on the Scope we use the span_id / trace_id
+    // To define the parent <-> child relationship
+    parentSpan = scope.getSpan() as Span;
     if (parentSpan) {
-      span = parentSpan.child(spanContext);
-      addedAsChild = true;
+      const { trace_id } = parentSpan.getTraceContext();
+      newSpanContext = {
+        traceId: trace_id,
+        ...spanOrTransactionContext,
+      };
     }
   }
 
-  if (!span) {
-    span = new Span(spanContext, hub);
-  }
+  // We are dealing with a Transaction
+  if ((newSpanContext as TransactionContext).name) {
+    const transaction = new Transaction(newSpanContext as TransactionContext, hub);
 
-  // We only roll the dice on sampling for "root" spans (transactions) because the childs inherit this state
-  if (span.sampled === undefined && span.isRootSpan()) {
-    const sampleRate = (client && client.getOptions().tracesSampleRate) || 0;
-    span.sampled = Math.random() < sampleRate;
-  }
+    // We only roll the dice on sampling for root spans of transactions because all child spans inherit this state
+    if (transaction.sampled === undefined) {
+      const sampleRate = (client && client.getOptions().tracesSampleRate) || 0;
+      transaction.sampled = Math.random() < sampleRate;
```

Comment:
`Math.random` is 0 inclusive. There's a chance that it'll evaluate to `false`, which we don't want. Use `<=` instead.